### PR TITLE
Fix HelmholtzDerivativeTaker

### LINKED ISSUE #107: test_translations fails when run with 3D kernels
Currently only 2D kernels are tested and when 3D kernels are tested the convergence
verifier fails.

### PR DESCRIPTION
**Pull request by isuruf**
Fixes #107 
Regression from #26 

Also adds a test for 3D kernels in test_translations.

### CODE REVIEW

**Comment by inducer** [comment]
Thanks!

File: sumpy/tools.py [hunk 1]
```diff
@@ -388,9 +389,11 @@ def diff(self, mi, q=0):
             expr /= self.r**2
         else:
             # See reference [Tausch2003] in RadialDerivativeTaker.diff
+            # Note that there is a typo in the paper where
+            # -k**2/r is given instead of -k**2/r**2.
```
Comment (inducer) [inline sumpy/tools.py:393]: Nice find! :tada:

File: test/test_kernels.py [hunk 2]
```diff
@@ -525,7 +525,10 @@ def test_translations(ctx_factory, knl, local_expn_class, mpole_expn_class):
 
     del eval_offset
 
-    orders = [3, 4, 5]
+    if knl.dim == 2:
+        orders = [2, 3, 4]
```
Comment (inducer) [inline test/test_kernels.py:529]: How come?

File: sumpy/tools.py [hunk 3]
```diff
@@ -392,8 +392,9 @@ def diff(self, mi, q=0):
             # Note that there is a typo in the paper where
             # -k**2/r is given instead of -k**2/r**2.
             k = (self.orig_expr * self.r).args[-1] / sym.I / self.r
-            expr = -(2*q - 1)/self.r**2 * self.diff(mi, q - 1)
-            expr += -k**2 / self.r**2 * self.diff(mi, q - 2)
+            expr = -(2*q - 1) * self.diff(mi, q - 1)
+            expr += -k**2 * self.diff(mi, q - 2)
+            expr /= self.r**2
```
Comment (inducer) [inline sumpy/tools.py:397]: Want to write this as a single expression? Less error-prone IMO.

File: test/test_kernels.py (hunk 2 above)
Comment (isuruf) [reply test/test_kernels.py:529]: 4->5 is problematic for most of 2D kernels like 2->3 is problematic for 3D kernels.

Comment (inducer) [reply test/test_kernels.py:529]: Fair enough. :shrug:

**Comment by inducer** [comment]
Thanks!

**Pull request merged**